code/game/objects/items.dm

```diff
@@ -631,6 +628,43 @@ GLOBAL_DATUM_INIT(fire_overlay, /mutable_appearance, mutable_appearance('icons/e
 
 	return M.can_equip(src, slot, disable_warning, bypass_equip_delay_self)
 
+/**
+ * This proc asks an item for special slot equipping priority before attempting to iterate through the slots.
+ * Arguments:
+ * * slot_priority: default list of slots to try and equip to in order. you take out and modify slots that it should conditionally not equip to.
+ */
+/obj/item/proc/special_slot_priority(list/slot_priority)
```

Comment:
Don't have this mutate the list. Have it return a new list instead (defaulting to `return slot_priority`). This makes it much easier to reason about consumers.